vectorisers.go

```diff
@@ -56,16 +52,15 @@ type RegExpTokeniser struct {
 	StopWords map[string]bool
 }
 
-// NewTokeniser returns a new, default Tokeniser implementation.  If
-// removeStopwords is true then stop words will be removed from tokens
-func NewTokeniser(removeStopwords bool) Tokeniser {
+// NewTokeniser returns a new, default Tokeniser implementation.
+// stopWords is a potentially empty string slice
+// that contains the words that should be removed from the corpus
+func NewTokeniser(stopWords []string) Tokeniser {
 	var stop map[string]bool
 
-	if removeStopwords {
-		stop = make(map[string]bool)
-		for _, word := range stopWords {
-			stop[word] = true
-		}
+	stop = make(map[string]bool)
+	for _, word := range stopWords {
```

Comment:
Can we allow people to supply a nil slice to all the New functions rather than an empty slice if they don't want to use stop words.  The same goes for Tokeniser New functions although I wonder if for those, we might consider adding an additional, alternative New function that allows clients to supply their own Tokeniser which can be constructed with their own stop words, overriding the default set.

Reply:
Where you thinking of a variadic function like: 
`func NewTokeniser(stopWords ...[]string) Tokeniser {`
in combination with the conditional `if (len(stopWord) != 0)`?

Reply:
Yup a variadic function would work.